guest/api/types.go

```diff
@@ -31,23 +31,24 @@ var _ FilterPlugin = FilterFunc(nil)
 // FilterFunc adapts an ordinary function to a FilterPlugin.
 type FilterFunc func(pod Pod, nodeInfo NodeInfo) *Status
 
-// Filter returns f(a).
+// Filter returns f(pod, nodeInfo).
 func (f FilterFunc) Filter(pod Pod, nodeInfo NodeInfo) *Status {
 	return f(pod, nodeInfo)
 }
 
-// ScorePlugin is a WebAssembly implementation of framework.ScorePlugin.
+// ScorePlugin is a WebAssembly implementation of framework.ScorePlugin except
+// the score is uint32, not int64.
```

Comment:
Looking at the PR I wonder if `int32` may be better that `uint32`. I have seen scoring situations before where default is zero, some things are higher priority than default and scores within positive range, others are lower priority than default but still scored within negative range. Whether this is something actually done in k8s I don't know but I guess matching the standard type semantically can prevent confusion, while restricting the precision because 64-bits is overkill and anyways most math in wasm will be 32-bits still shouldn't be too confusing / limiting.

Anyways mostly an observation as food for thought, definitely not an expert on it :)

Reply:
In theory, yes, we have no idea how people design their plugins. Although after normalizing, they will be in the range of [0, 100]

Reply:
No strong opinions. I made it uint32 randomly, I admit it ;) I'll move to int32, and also add a test that negative is visible.

Reply:
Yes, even in in-tree plugins, I remember some of them return negative scores in Score and then make it in [0, 100] in NormalizeScore(). So, int32 LGTM